homeassistant/components/synology_dsm/__init__.py

```diff
@@ -401,71 +431,72 @@ def subscribe(self, api_key, unique_id):
         @callback
         def unsubscribe() -> None:
             """Unsubscribe an entity from API fetches (when disable)."""
+            _LOGGER.debug(
+                "SynoAPI.unsubscribe() - api_key:%s, unique_id:%s", api_key, unique_id
+            )
             self._fetching_entities[api_key].remove(unique_id)
+            if len(self._fetching_entities[api_key]) == 0:
+                self._fetching_entities.pop(api_key)
 
         return unsubscribe
 
     @callback
-    def _async_setup_api_requests(self):
+    def _setup_api_requests(self):
```

Comment:
Why did we change the name? We prefer naming functions that should run in the event loop `async_*`.